### PR TITLE
Build ci gates per service

### PR DESCRIPTION
**Pull request by vhvb1989**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by vhvb1989** [comment]
## Results:

### cpp - identity
Windows  - [1:15](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314494&view=logs&j=e71eac3d-89c8-52e0-61cd-202dc96cdaa6&t=b6104d90-d336-5174-f8e1-130f65c2cb81&l=9) v/s [3:29](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314258&view=logs&j=a11ac2a6-568a-5932-fa40-30c4f3126c80&t=045fb480-78fb-5cc9-1875-4bdf010d64f7&l=11)
Linux - [41 s](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314494&view=logs&j=ca6c228c-7ba2-55ec-3482-b320fca07ebb&t=844ea3b0-8adc-5acf-5faf-db87074ec22a&l=9) v/s [ 4:51 min ](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314258&view=logs&j=ca6c228c-7ba2-55ec-3482-b320fca07ebb&t=19bdfdfd-dade-56b3-ac4c-44e9b5556c81&l=11)
Mac - [2:14]( https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314494&view=logs&j=e109b754-e5a4-58aa-96d5-f28daf32bc3c&t=bd849219-bc00-54c1-16d3-e00aa95f6786&l=9 ) min v/s [12:16 min ](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1314258&view=logs&j=e109b754-e5a4-58aa-96d5-f28daf32bc3c&t=ba4898b6-8192-5319-125f-11f9fed782cb&l=11)

**Comment by vhvb1989** [comment]
image

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,16 +13,17 @@ option(BUILD_TRANSPORT_CURL "Build an HTTP transport implementation with CURL" O
 option(BUILD_TRANSPORT_WINHTTP "Build an HTTP transport implementation with WIN HTTP" OFF)
 option(BUILD_TRANSPORT_CUSTOM "Implementation for AzureSdkGetCustomHttpTransport function must be linked to the final application" OFF)
 option(BUILD_TESTING "Build test cases" OFF)
+option(BUILD_SAMPLES "Build test cases" OFF)
```
Comment (RickWinter) [inline CMakeLists.txt:16]:
```suggestion
option(BUILD_SAMPLES "Build sample application for Azure Storage clients" OFF)
```

File: sdk/identity/azure-identity/CMakeLists.txt [hunk 2]
```diff
@@ -93,12 +95,13 @@ if(BUILD_TESTING)
...
 endif()
 
-if (AZ_ALL_LIBRARIES)
+if (BUILD_SAMPLES)
   add_subdirectory(samples)
 endif()
+
+if (BUILD_PERFORMANCE_TESTS)
+  add_subdirectory(test/perf)
+endif()
```
Comment (RickWinter) [inline sdk/identity/azure-identity/CMakeLists.txt:107]:
```suggestion
endif()

```

**Review by RickWinter — APPROVED** [review]

**Pull request merged**